_summaries/sukhbaatar2015end-to-end_1.md

```diff
@@ -1,19 +1,76 @@
 ---
 layout: summary
 title: Summary
-paper: {{paper_tag}}
+paper: sukhbaatar2015end-to-end
 # Please fill out info below
-author: # Your GitHub id
-score: # How did you like this paper 0(dislike) to 10(love)
+author: DartingMelody
+score: 7
 ---
 
-TODO: Summarize the paper:
-* What is the core idea?
-* How is it realized (technically)?
-* How well does the paper perform?
-* What interesting variants are explored?
+**What is the core idea?**
+
+The paper introduces a novel recurrent neural network with an explicit large memory and a recurrent attention mechanism. The model reads recurrently from memory multiple times before giving the results. This model is trained end-to-end and therefore requires comparitively less supervision during training. Hence, it can also be applied to realistic settings. The paper also shows experimentally that multiple hops over the long-term memory leads to good performance of the model. 
+
+**How is it realized (technically)?**
+
+The Model takes as input $$x_{1},...,x_{i}$$ (to store in memory), query _q_ and outputs answer _a_. The model writes all _x_ to the memory up to a fixed buffer size and creates continuous representation of _x_ and _q_. 
+
+*Single layer:*
+
+1) The input set{$$x_{i}$$} is embedded with matrix _A_ ( _d_ x _V_) to give memory vectors {$$m_{i}$$}. 
+2) The query _q_ is also embedded with matrix _B_ (_d_ x _V_) to give an internal state u.
+3) On computing match between each memory {$$m_{i}$$} and _u_ by taking softmax of their inner product which results into the probability vector p over the inputs i.e. $$p_{i} = Softmax(u^{T}m_{i})$$
+4) Each $$x_{i}$$ has a corresponding mapping to an ouput vector $$c_{i}$$ (using another embedding matrix _C_).
+5) The final output is the sum of the weighted probabilities {$$p_{i}$$}  over the transformed input $$c_{i}$$ i.e. $$o = \sum_{i}^{}p_{i}c_{i}$$
```

Comment:
super small note but cI is not a transformed input, it's an embedded memory of the network. qi is the weighted vector from the input which I think is what you were referencing?  (Super small distinction, just caught my eye)

Reply:
This is actually transformed inputs ci (the paper has also mentioned that).